File: back-end/src/main/java/ma/infosat/gestion/adherent/entities/BureauVote.java
package ma.infosat.gestion.adherent.entities;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonSetter;

@Entity
public class BureauVote implements Serializable {
	
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long bureauVote_id;
	private  String nom_bureauVote;
	
	private String adresse;
	
	@OneToMany(mappedBy="bureauVote")
	private Set<Telephone> telephones=new HashSet<Telephone>();
	
	@OneToMany(mappedBy="bureauVote",cascade=CascadeType.ALL)
	private Set<Adherent> adherents =new HashSet<Adherent>();

	public BureauVote(String nom_bureauVote,String adresse) {
		super();
		this.nom_bureauVote = nom_bureauVote;
		this.adresse=adresse;
	}
	
	public BureauVote() {}

	public String getNom_bureauVote() {
		return nom_bureauVote;
	}

	public void setNom_bureauVote(String nom_bureauVote) {
		this.nom_bureauVote = nom_bureauVote;
	}

	public String getAdresse() {
		return adresse;
	}

	public void setAdresse(String adresse) {
		this.adresse = adresse;
	}
   @JsonIgnore
	public Set<Telephone> getTelephones() {
		return telephones;
	}
    @JsonSetter
	public void setTelephones(Set<Telephone> telephones) {
		this.telephones = telephones;
	}
    @JsonIgnore
	public Set<Adherent> getAdherents() {
		return adherents;
	}
    @JsonSetter
	public void setAdherents(Set<Adherent> adherents) {
		this.adherents = adherents;
	}

	public Long getBureauVote_id() {
		return bureauVote_id;
	}

}

File: front-end/src/app/entities/bureau-vote.ts
import { Telephone } from './telephone';
export class BureauVote {

  private _telephones:Telephone[];

  constructor(private _bureauVote_id?:Number,private _nom_bureauVote ?:string,
              private _adresse ?:string
  ) {
  }
  get bureauVote_id():Number{
    return  this._bureauVote_id;
  }
  set bureauVote_id(value:Number){
    this._bureauVote_id=value;
  }
  get nom_bureauVote(){
    return this._nom_bureauVote;
  }
  set nom_bureauVote(value:string){
    this._nom_bureauVote=value;
  }
  get adresse(){
    return this._adresse;
  }
  set adresse(value:string){
    this._adresse=value;
  }

  get telephones():Telephone[]{
    return this._telephones;
  }
  set telephones(value:Telephone[]){
    this._telephones=value;
  }
}
